### PR TITLE
imaginary - fix autorotate for heic and improve the logic

### LINKED ISSUE #35643: Imaginary preview generation is not working as expected for folders containing HEIC images.
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

### Steps to reproduce
1. Install imaginary plugin for preview generation
2. Create a folder with several (tested with ~110) HEIC images (Images shot on a Samsung A52s phone with the stock camera app)
3. Try to open/preview that folder in the Web-UI

### Expected behavior 
Images with accompanying previews should appear.

### Actual behavior 
The imaginary binary from the imaginary docker container commits all available memory and swap space, finally getting killed a few minutes in after running out of memory.
Preview generations succeed very slowly but it seems that already generated previews do not get used successfully, at least I could never get them to show up.

It is notable that the previews for other files in the folder (.mp4 videos, 4 .jpg images) work and appear.

### Host OS 
Ubuntu 20.04.5 LTS (VPS with 4vCPUs, 8GB of RAM and 2GB configured swap)

#### Nextcloud AIO version 
Nextcloud AIO v3.1.0, Nextcloud 24.0.7

#### Current channel 
latest

#### Other valuable info 
Relevant link with a bug report of another user: https://help.nextcloud.com/t/imaginary-not-working-on-nextcloud-aio/146725

Logfile of the imaginary container: [aio_imaginary_log.txt](https://github.com/nextcloud/all-in-one/files/10171468/aio_imaginary_log.txt)

Screenshots of htop on the VPS
htop-screenshot-1
htop-screenshot-2
htop-screenshot-3

### PR DESCRIPTION
**Pull request by szaimen**
Fix https://github.com/nextcloud/server/issues/35643

### CODE REVIEW

**Review by artonge — APPROVED** [review]
Fine beside question

File: lib/private/Preview/Imaginary.php [hunk 1]
```diff
@@ -92,50 +98,43 @@ public function getCroppedThumbnail(File $file, int $maxX, int $maxY, bool $crop
...
-					'params' => [
-						'width' => $maxX,
-						'height' => $maxY,
-						'stripmeta' => 'true',
-						'type' => $mimeType,
-						'norotation' => 'true',
-						'quality' => $quality,
-					]
-				]
+		} elseif ($autorotate) {
```
[the remark this answers is not in the document]
Comment (szaimen) [reply lib/private/Preview/Imaginary.php:118]: Currently they are as autorate only makes sense for images that dont need to be converted and the other way around.

**Review by come-nc — APPROVED** [review]

**Comment by loeffelpan** [comment]
Thanks for backporting this.
But is it really fixed as autorotation is just disables for HEIC?
Any thoughts how make autorotation work for HEIC?

**Pull request merged**